src/include/simeng/pipeline/ReorderBuffer.hh

```diff
@@ -85,6 +89,10 @@ class ReorderBuffer {
   /** Get the number of speculated loads which violated load-store ordering. */
   uint64_t getViolatingLoadsCount() const;
 
+  /** Get the value of the PC that should come after the last committed
+   * instruction. */
+  uint64_t getNextPC() const { return nextPC_; };
```

Comment:
I think the only viable option is to move towards the solution of context switching imidately and flushing the pipeline manually. Can look into this if desirable.

Otherwise, given the surrounding logic, saving `nextPC_` seems insignificant overhead

Reply:
See https://github.com/UoB-HPC/SimEng/pull/284#discussion_r1106038263